Arrowgene.Ddon.GameServer/Handler/PawnGetPartyPawnDataHandler.cs

```diff
@@ -24,13 +21,19 @@ public PawnGetPartyPawnDataHandler(DdonGameServer server) : base(server)
             _OrbUnlockManager = server.OrbUnlockManager;
         }
 
-        public override void Handle(GameClient client, StructurePacket<C2SPawnGetPartyPawnDataReq> packet)
+        public override S2CPawnGetPartyPawnDataRes Handle(GameClient client, C2SPawnGetPartyPawnDataReq packet)
         {
             // var owner = Server.CharacterManager.SelectCharacter(packet.Structure.CharacterId);
-            GameClient owner = this.Server.ClientLookup.GetClientByCharacterId(packet.Structure.CharacterId);
+            GameClient owner = this.Server.ClientLookup.GetClientByCharacterId(packet.CharacterId);
             // TODO: Move this to a function or lookup class
             List<Pawn> pawns = owner.Character.Pawns.Concat(client.Character.RentedPawns).ToList();
-            Pawn pawn = pawns.Where(pawn => pawn.PawnId == packet.Structure.PawnId).First();
+
+            if (!pawns.Any())
+            {
+                throw new ResponseErrorException(ErrorCode.ERROR_CODE_PAWN_NOT_FOUNDED);
+            }
+
+            Pawn pawn = pawns.Where(pawn => pawn.PawnId == packet.PawnId).First();
```

Comment:
Shouldnt this throw if it can't find a pawn with a matching id and not if it can't find any pawn at all?